src/blockdata/script.rs

```diff
@@ -773,22 +789,21 @@ impl<'a> Iterator for Instructions<'a> {
                 // casting is safe because we don't support 16-bit architectures
                 let n = n as usize;
 
-                if self.data.len() < n {
-                    let data_len = self.data.len();
-                    self.data.nth(data_len); // Kill iterator so that it does not return an infinite stream of errors
-                    return Some(Err(Error::EarlyEndOfScript));
-                }
-                if self.enforce_minimal {
-                    // index acceess is safe because we checked the lenght above
-                    if n == 1 && (self.data.as_slice()[0] == 0x81 || (self.data.as_slice()[0] > 0 && self.data.as_slice()[0] <= 16)) {
-                        let data_len = self.data.len();
-                        self.data.nth(data_len); // Kill iterator so that it does not return an infinite stream of errors
-                        return Some(Err(Error::NonMinimalPush));
+                let op_byte = self.data.as_slice().first();
```

Comment:
nit:
This is not the `op_byte`. This is more like `first_byte` and we are checking `first_byte` is not equal to `op_byte`. 

```rust
    fn next(&mut self) -> Option<Result<Instruction<'a>, Error>> {
        let &byte = self.data.next()?; // this should be op_byte
```